--- a/novadocker/virt/docker/driver.py
+++ b/novadocker/virt/docker/driver.py
@@ -260,14 +260,24 @@
         #
         #                   Also see:
         #                    docker/docs/sources/articles/runmetrics.md
-        info = hardware.InstanceInfo(
-            max_mem_kb=mem,
-            mem_kb=mem,
-            num_cpu=num_cpu,
-            cpu_time_ns=0,
-            state=(power_state.RUNNING if running
-                   else power_state.SHUTDOWN)
-        )
+        try:
+            info = hardware.InstanceInfo(
+                max_mem_kb=mem,
+                mem_kb=mem,
+                num_cpu=num_cpu,
+                cpu_time_ns=0,
+                state=(power_state.RUNNING if running
+                       else power_state.SHUTDOWN)
+            )
+        except:
+            info = dict(
+                max_mem_kb=mem,
+                mem_kb=mem,
+                num_cpu=num_cpu,
+                cpu_time_ns=0,
+                state=(power_state.RUNNING if running
+                       else power_state.SHUTDOWN)
+            )
         return info
 
     def get_host_stats(self, refresh=False):
